Code/Framework/AzCore/AzCore/Memory/PoolAllocator.cpp

```diff
@@ -29,6 +28,29 @@
 #define POOL_ALLOCATION_MIN_ALLOCATION_SIZE size_t{8}
 #define POOL_ALLOCATION_MAX_ALLOCATION_SIZE size_t{512}
 
+namespace
+{
+    struct Magic32
+    {
+        static const AZ::u32 m_defValue = 0xfeedf00d;
+        AZ_FORCE_INLINE Magic32()
+        {
+            m_value = (m_defValue ^ (AZ::u32)((AZStd::size_t)this));
```

Comment:
The `this` pointer should be cast using `uintptr_t` as that is the appropriate type for casting between pointer and integer types.
`size_t` is not an appropriate type.
```suggestion
            m_value = (m_defValue ^ AZ::u32(reinterpret_cast<AZStd::uintptr_t>(this));
```